### PR TITLE
docs: document mfe patches catalog

### PR DESCRIPTION
**Pull request by Danyal-Faheem**
**Changes**

- Added a catalog which includes every patch available for tutor-mfe in the README.
- Also included a one liner description of each patch for users to understand.

Fixes this [issue.](https://github.com/overhangio/tutor-mfe/issues/168)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -365,11 +371,101 @@ You will also have to manually remove a few settings::
...
+
+mfe-lms-common-settings
+~~~~~~~~~~~~~~~~~~~~~~~
+Python-formatted LMS settings used both in production and development.
+
+File changed: ``apps/openedx/settings/partials/common_lms.py``
+
+mfe-webpack-dev-config
+~~~~~~~~~~~~~~~~~~~~~~
+Add any additional configurations to be added at the end of the development webpack config file in Javascript format.
```
Comment (regisb) [inline README.rst:408]: "Add" and "additional" are redundant. I suggest you replace "Add any additional XYZ" by "Add any XYZ" everywhere.

File: README.rst [hunk 2]
```diff
@@ -365,11 +371,101 @@ You will also have to manually remove a few settings::
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Add any additional instructions for before the npm install is initiated.
+
+File changed: ``tutormfe/templates/mfe/build/mfe/Dockerfile``
+
+mfe-dockerfile-pre-npm-install-{}
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Add any additional instructions for before the npm install is initiated for a specific mfe. Add the exact mfe name at the end to only change instructions for that mfe.
+
+Example: ``mfe-dockerfile-pre-npm-install-learning`` Will only apply any instructions specified for the learning mfe.
```
Comment (regisb) [inline README.rst:422]: Will -> will

File: README.rst [hunk 3]
```diff
@@ -365,11 +371,101 @@ You will also have to manually remove a few settings::
...
+
+mfe-dockerfile-pre-npm-build
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Add any additional instructions for before the build step initializes. This will apply the instructions to every MFE. For an example on the usage of this patch, see `over here <#mfe-docker-pre-npm-build>`_.
+
+File changed: ``tutormfe/templates/mfe/build/mfe/Dockerfile``
+
+mfe-dockerfile-pre-npm-build-{}
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Add any additional instructions for before the build step initializes for a specific mfe. Add the exact mfe name at the end to only change instructions for that mfe. For an example on the usage of this patch, see `over here <#mfe-docker-pre-npm-build>`_.
```
Comment (regisb) [inline README.rst:448]: a specific mfe -> a specific MFE
for that mfe -> for that MFE

(same for other occurrences of "mfe" in the document)

File: changelog.d/20231212_131405_danyal.faheem_document_mfe_patches_catalog.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+
+- [Improvement] Added a patch catalog for tutor-mfe which includes a one liner description of every patch available. (by @Danyal-Faheem)
```
Comment (regisb) [inline changelog.d/20231212_131405_danyal.faheem_document_mfe_patches_catalog.md:2]: No need to add a changelog entry, as this change is not affecting the end user.

**Review by regisb — CHANGES_REQUESTED** [review]
This is excellent work! I only have a few minor comments.

**Comment by regisb** [comment]
@arbrandes what do you think?

**Review by arbrandes — APPROVED** [review]
This is fantastic!  Thanks!

**Pull request merged**